packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx

```diff
@@ -37,6 +38,10 @@ interface TooltipPlugin2Props {
   // y-only, via shiftKey
   clientZoom?: boolean;
 
+  onSelect?: OnSelectCallback;
+  // Whether to clear or keep the selected area rendered after the user has finished dragging
+  keepSelectedArea?: boolean;
```

Comment:
maybe we can skip this and instead `return false` from `onSelect` callback to effectively `preventDefault()`.

or instead of returning `false` we can instead return an `enum` value so the callback can dynamically determine if the selection should be cleared or not.

an/or we can pass a `clearSelection` callback to the `onSelect` callback that can be invoked to hide the selection (maybe once the querying is done), but i'm a bit worried about memory leaks with this API.